resources/providers/awslib/ec2_metadata_provider.go

```diff
@@ -42,5 +42,5 @@ func (provider Ec2MetadataProvider) GetMetadata(ctx context.Context, cfg aws.Con
 		return ec2imds.GetInstanceIdentityDocumentOutput{}.InstanceIdentityDocument, err
 	}
 
-	return identityDocument.InstanceIdentityDocument, err
+	return identityDocument.InstanceIdentityDocument, nil
```

Comment:
Is it related to this PR?

Reply:
No.
It does not make sense to return the `err` that is nil, instead, return nil.
Note that this error was checked above.